experimental/Schemes/WeilDivisor.jl

```diff
@@ -0,0 +1,352 @@
+export WeilDivisor
+export scheme_type, ideal_sheaf_type, coefficient_ring_type, coefficient_type
+export scheme, components, coefficient_dict, coefficient_ring
+export in_linear_system
+
+export LinearSystem 
+export divisor, find_subsystem
+
+@Markdown.doc """
+    WeilDivisor
+
+A Weil divisor on a covered scheme ``X``; 
```

Comment:
Following Hartshorne we want `X` to be a noetherian integral separated scheme which is regular in codimension one.

Reply:
@simonbrandhorst: That information should end up in the .md files. Do we want to mention it here, potentially doubling something from the introductory text to the Weil divisors (once it is written)?

Reply:
There are several viable ways to write the documentation. 
One way is to attach (part of) the intro to the definition of the type 
and import that in the .md file. This has the advantage that typing `?WeilDivisor` returns something useful.
Another option is the one you mention.